packages/docusaurus-theme-search-algolia/src/index.js

```diff
@@ -14,5 +14,25 @@ module.exports = function() {
     getThemePath() {
       return path.resolve(__dirname, './theme');
     },
+
+    configureWebpack() {
+      // Ensure that algolia docsearch css is its own chunk
+      return {
+        optimization: {
+          splitChunks: {
+            cacheGroups: {
+              algolia: {
+                name: 'algolia',
+                test: /algolia\.css$/,
+                chunks: `all`,
+                enforce: true,
+                // Set priority higher than docusaurus single-css extraction
+                priority: 60,
```

Comment:
Magic number which will probably break in future. `z-index` has the same issue and the solution is to have a centralized list of `z-index` which various CSS files can refer to. But I don't have a good solution here.